docs/source/sci_api_tutorial/preparation_protonate_stru.rst

```diff
@@ -0,0 +1,178 @@
+
+Briefs
+==============================================
+
+This science API, named ``enzy_htp.preparation.protonate.protonate_stru``,
+solves the protein protonation problem that add missing H atoms to the supplied
+``enzy_htp.structure.Structure`` class instance (hereafter referred to as ``Structure`` instance).
+Protonation states are determined for residues with multiple ones.
+
+Input/Output
+==============================================
+
+**input**: A ``Structure`` instance (no matter it's a protein, polypeptite, or ligand).
+
+.. dropdown:: :fa:`eye,mr-1` How to obtain ``Structure`` instance?
+
+    Structure can be obtained by 
+    
+    1. parsing from a file using one of the StructureParser:
+
+    - `PDBParser <xxx>`_
+    - `PrmtopParser <xxx>`_
+
+    2. OR using the output of `Remove Solvent <preparation_remove_solvent.html>`_ or `Remove Hydrogens <preparation_remove_solvent.html>`_. (Commonly used here)
+
+**output**: A ``Structure`` instance of protonated structure (in-place modification, not as return value).
+
+.. panels::
+
+    :column: col-lg-12 col-md-12 col-sm-12 col-xs-12 p-2 text-left
+
+    .. image:: ../../figures/preparation_protonate_stru_dfd.svg
+        :width: 100%
+        :alt: preparation_remove_solvent
+
+Arguments
+==============================================
+
+``stru``
+    The input ``Structure`` instance (no matter it's a protein, polypeptite, or ligand).
+
+``ph``
+    The pH value for determining the protonation state.
+    (Float, optional, default ``7.0``)
+
+    .. dropdown:: :fa:`eye,mr-1` Click to learn more about ``ph``
+
+        The choice of pH value depends on the physiological environment of the protein you are simulating.
+
+        For example, if you are simulating a protein in human blood, then the pH value is best set in the range [7.35, 7.45].
+
+``protonate_ligand``
+    If also protonate ligand.
+    
+    (Boolean, optional, default ``True``)
+
+``engine``
+    Engine for determining the pKa and adding hydrogens to the protein peptide part.
+
+    (String, optional, default ``pdb2pqr``) 
+    
+    .. dropdown:: :fa:`eye,mr-1` Click to learn more about ``engine``
+
+        The ``engine`` option determines which third-party tool you use to add hydrogen atoms to your protein (polypeptide).
+
+        Currently only ``pdb2pqr`` is available. More ``engine`` options will be available in future versions.
+
+``ligand_engine``
+    Engine for adding hydrogens to ligands (current available values: ``pybel``)
+
+    (String, optional, default ``pybel``)
+    
+    .. dropdown:: :fa:`eye,mr-1` Click to learn more about ``engine``
+
+        The ``ligand_engine`` option determines which third-party tool you use to add hydrogen atoms to your ligand.
+
+        Currently only ``pybel`` is available. More ``ligand_engine`` options will be available in future versions.
+
+``**kwarg``
+    Setting(s)/Option(s) related to specific engine.
+
+    (optional, default ``None``)
```

Comment:
replace this with a section that contain kwargs when engine="pdb2pqr" (check the code)

Reply:
Replaced by 3 conditional arguments.